_posts/2022-02-06-Makefile_to_create_githook.md

```diff
@@ -0,0 +1,60 @@
+---
+title: Using Makefile to Create Git Hook
+toc: true
+layout: post
+categories: [technical]
+author: Hiromi Suenaga
+hide: false
+---
+
+In the [previous post](https://blog.problemsolversguild.com/technical/2022/02/05/GitHook_to_clean_notebook.html), we created a Git Hook that will stirip all the superfluous metadata automatically when you commit notebooks. We wanted to make the installation of this hook easier by creating a Makefile target.
+
+Here is what we came up with: 
+
+```bash
+SHELL := /bin/bash
+
+.PHONY: hook
+hook: 
+	@echo -e '#!/bin/sh\nfor file in $$(git diff --diff-filter=d --cached --name-only | grep -E '"'"'\.ipynb$$'"'"')\ndo\n\tjupyter nbconvert --ClearOutputPreprocessor.enabled=True --inplace "$$file"\n\tgit add "$$file"\ndone\n'  > .git/hooks/pre-commit
+	@chmod +x .git/hooks/pre-commit
+```
+[[Download]('https://blog.problemsolversguild.com/assets/Makefile')]
+
+Run `make hook` and Voila! You have your Git Hook that will reduce merge conflicts headaches.
+
+---
+
+## My web searches while creating this Makefile
+
+### How to convert multiline file into a string with newline `\n` characters:
+```bash
+awk '$1=$1' ORS='\\n' file
+```
+>This is how I converted an existing hook file into a single string. I had to add tabs (`\t`) manually, but this `awk` got me close enough. 
```

Comment:
Where did the original hook file come from?  Are those installed by default?

Reply:
Maybe an example here of using this on your hook file would be nice

Reply:
Updated

Reply:
Original hook file is something we created in the previous post.